### PR TITLE
[Connectivity] wifi removal

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

This PR removes all the wifi related APIs in the connectivity plugin. It is a breaking change. Since 0.4.y is compatible with 1.0.0 we're bumping the version straight to 2.0.0

We introduced wifi_info_flutter and all the wifi related APIs are moved to wifi_info_flutter.

We will keep the connectivity_platform_interface as is until the desktop plugin and web plugin are migrated to wifi_info_flutter as well.

**We should land this after wifi_info_flutter has published**

## Related Issues

Fixes https://github.com/flutter/flutter/issues/40418

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by blasten** [comment]
Offline discussion: This change will be merged to the nnbd branch, so there's no need for `2.0.0`, then `3.0.0-nullsafety`.

@cyanglaz what about closing this PR, and reopening one for the nnbd branch?

File: packages/connectivity/connectivity/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+## 2.0.0
+
+* [Breaking Change] The `getWifiName`, `getWifiBSSID` and `getWifiIP` are removed to [wifi_info_flutter](https://github.com/flutter/plugins/tree/master/packages/wifi_info_flutter)
+* Migration guide:
+
+  If you don't use any of the above APIs, your code should work as is. In addition, you can also remove `NSLocationAlwaysAndWhenInUseUsageDescription` and `NSLocationWhenInUseUsageDescription` in `ios/Runner/Info.plist`
+
+  If you use any of the above APIs, you can find the same APIs in the [wifi_info_flutter](https://github.com/flutter/plugins/tree/master/packages/wifi_info_flutter/wifi_info_flutter) plugin.
+  For example, to migrate `getWifiName` to use the new plugin:
```
Comment (blasten) [inline packages/connectivity/connectivity/CHANGELOG.md:9]:
```suggestion
  For example, to migrate `getWifiName`, use the new plugin:
```

**Review by blasten — APPROVED** [review]
LGTM

**Review by bparrishMines — APPROVED** [review]
LGTM

**Comment by brunobowden** [comment]
@cyanglaz - thanks for landing this. I think going to v2.0.0 makes sense. I'll start making use of it.

**Pull request merged**